internal/pkg/aws/ssm/ssm.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ssm provides a client to make API requests to Amazon Systems Manager.
+package ssm
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+type api interface {
+	PutParameter(input *ssm.PutParameterInput) (*ssm.PutParameterOutput, error)
+}
+
+type SSM struct {
+	client api
+}
+
+// New returns a SSM service configured against the input session.
+func New(s *session.Session) *SSM {
+	return &SSM{
+		client: ssm.New(s),
+	}
+}
+
+// PutSecretInput contains fields needed to create or update a secret.
+type PutSecretInput struct {
+	Name      string
+	Value     string
+	Overwrite bool
+	Tags      map[string]string
+}
+
+// PutSecretOutput wraps an ssm PutParameterOutput struct.
+type PutSecretOutput ssm.PutParameterOutput
+
+// PutSecret creates or updates a SecureString parameter.
+func (s *SSM) PutSecret(in PutSecretInput) (*PutSecretOutput, error) {
```

Comment:
hmm maybe we don't need the output for `PutParameter`? I can hardly think of a use case.

Reply:
Should the decision be left for the caller? That is if the caller decides they don't need it, they could just do `_, err := PutParameter(in)`.

The output has the `Version` information, which might... perhaps... possibly... be useful.